ldapauthenticator/ldapauthenticator.py

```diff
@@ -463,17 +462,12 @@ async def authenticate(self, handler, data):
             username, resolved_dn = self.resolve_username(username)
             if not username:
                 return None
-            if str(self.lookup_dn_user_dn_attribute).upper() == "CN":
-                # Only escape commas if the lookup attribute is CN
-                username = re.subn(r"([^\\]),", r"\1\,", username)[0]
             if not bind_dn_template:
                 bind_dn_template = [resolved_dn]
 
         is_bound = False
         for dn in bind_dn_template:
-            userdn = dn.format(username=username)
-            if self.escape_userdn:
-                userdn = escape_filter_chars(userdn)
+            userdn = dn.format(username=escape_rdn(username))
```

Comment:
escape_filter_chars and escape_rdn seem to escape for different things unless I'm missing something. 

```
escape_filter_chars(r"\foo + , (N") -> '\\5cfoo + , \\28N'
escape_rdn(r"\foo + , (N") -> '\\\\foo \\+ \\, (N'
```

Is it ok to exchange them here?

Reply:
I've added a comment to this in another PR merged after this, linking to an RFC

Reply:
I guess you are correct. You explained this at the beginning of the PR but first I didn't understand this.